lib/src/main/java/org/esbtools/eventhandler/NotificationRepository.java

```diff
@@ -21,16 +21,35 @@
 import java.util.Collection;
 import java.util.List;
 
+/**
+ * Abstracts a transactional backing store and staging area for {@link Notification notifications}.
+ *
+ * <p>A repository is responsible for handling CRUD and data parsing operations around these
+ * objects, in whatever scheme necessary.
+ *
+ * <p>Production implementations are expected to be thread safe, even across a network.
+ */
 public interface NotificationRepository {
 
     /**
-     * Retrieves the oldest {@code maxEvents} {@link Notification notifications}, oldest first.
+     * Retrieves the oldest {@code maxNotifications} {@link Notification notifications}, oldest
+     * first.
      *
      * <p>Once retrieved, a notification should not be retrieved again, atomically. That is, many
-     * threads looking up notifications at the same time should all et a unique non-overlapping sample
-     * of the oldest notifications. Subsequent calls should always return a unique set.
+     * threads looking up notifications at the same time should all et a unique non-overlapping
+     * sample of the oldest notifications. Subsequent calls should always return a unique set.
```

Comment:
Yeah, it is a similar thing with priority document events. I want to do random but has to weight older notifications / higher priority document events. For now I am (or will be, in the case of notifications) just going down the list and locking in order.

Also, I wanted to ask for your thoughts on what to lock for notifications. We could do by identity like document events, which would open up opportunity to similarly check for superseded notifications, but currently we don't do this, and the APIs aren't there to do this. I'm leaning towards just locking by entity name or specific notification id for now and seeing how that goes.